app/javascript/mastodon/components/media_gallery.jsx

```diff
@@ -87,14 +95,18 @@ class Item extends PureComponent {
     let badges = [], thumbnail;
 
     let width  = 50;
-    let height = 100;
+    let height = 50;
 
-    if (size === 1) {
-      width = 100;
-    }
+    const cols = colCount(size);
+    const remaining = (-size % cols + cols) % cols;
+    const largeCount = Math.floor(remaining / 3); // width=2, height=2
```

Comment:
I wonder if there's a way to do this more responsively, and with less math?

Reply:
Similar to masonly, This requires some calculation to make grid layout. css grid cannot do that yet.